pkg/plugin/linuxutil/ethtool_stats_linux_test.go

```diff
@@ -69,32 +74,71 @@ func TestReadInterfaceStats(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
+			name: "test unsported interface",
+			opts: &EthtoolOpts{
+				errOrDropKeysOnly: false,
+				addZeroVal:        false,
+			},
+			statsReturn: nil,
+			statErr:     errInterfaceNotSupported,
+			result:      nil,
+			wantErr:     false,
+		},
+		{
+			name: "test skipped interface",
+			opts: &EthtoolOpts{
+				errOrDropKeysOnly: false,
+				addZeroVal:        false,
+			},
+			statsReturn: nil,
+			statErr:     errInterfaceNotSupported,
+			result:      nil,
+			wantErr:     false,
+		},
 	}
 
 	for _, tt := range tests {
 		l.Infof("Running TestReadInterfaceStats %s", tt.name)
+
 		ctrl := gomock.NewController(t)
 		defer ctrl.Finish()
 
 		ethHandle := NewMockEthtoolInterface(ctrl)
 		ethReader := NewEthtoolReader(tt.opts, ethHandle)
+
+		ethReader.unsupported = globalUnsupportedCache
+
 		assert.NotNil(t, ethReader)
 
-		ethHandle.EXPECT().Stats(gomock.Any()).Return(tt.statsReturn, nil).AnyTimes()
+		ethHandle.EXPECT().Stats(gomock.Any()).Return(tt.statsReturn, tt.statErr).AnyTimes()
 		ethHandle.EXPECT().Close().Times(1)
 		InitalizeMetricsForTesting(ctrl)
 
-		testmetric := prometheus.NewGauge(prometheus.GaugeOpts{
-			Name: "testmetric",
-			Help: "testmetric",
-		})
+		if tt.statErr == nil {
+			testmetric := prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "testmetric",
+				Help: "testmetric",
+			})
 
-		MockGaugeVec.EXPECT().WithLabelValues(gomock.Any()).Return(testmetric).AnyTimes()
+			MockGaugeVec.EXPECT().WithLabelValues(gomock.Any()).Return(testmetric).AnyTimes()
+		}
 
 		err := ethReader.readInterfaceStats()
 		assert.Nil(t, err)
 
-		ethReader.updateMetrics()
+		if tt.statErr == nil {
+			ethReader.updateMetrics()
+		}
+
+		if tt.statErr != nil && errors.Is(tt.statErr, errInterfaceNotSupported) {
+			assert.NotEqual(t, nil, ethReader.unsupported, "unsupported map should not be nil")
+			assert.NotEmpty(t, ethReader.unsupported, "unsupported map should contain interfaces")
+		}
+
+		globalUnsupportedCache = ethReader.unsupported
+		fmt.Println("Current unsupported cache: ", ethReader.unsupported, "Global unsupported cache: ", globalUnsupportedCache)
```

Comment:
Generally, you shouldn't use `fmt.Println` in tests. `t.Log` is better since the output is more orderly.